sway/desktop/fx_renderer/fx_framebuffer.c

```diff
@@ -1,9 +1,16 @@
+#include <wlr/util/box.h>
 #include "log.h"
 #include "sway/desktop/fx_renderer/fx_framebuffer.h"
 
-void fx_framebuffer_bind(struct fx_framebuffer *buffer, GLsizei width, GLsizei height) {
+void fx_framebuffer_bind(struct fx_framebuffer *buffer, struct wlr_output *output) {
+	struct wlr_box monitor_box;
+	wlr_output_transformed_resolution(output, &monitor_box.width, &monitor_box.height);
+	wlr_box_transform(&monitor_box, &monitor_box,
+			wlr_output_transform_invert(output->transform),
+			monitor_box.width, monitor_box.height);
+
 	glBindFramebuffer(GL_FRAMEBUFFER, buffer->fb);
-	glViewport(0, 0, width, height);
+	glViewport(0, 0, monitor_box.width, monitor_box.height);
 }
```

Comment:
we already need to get the width and height in the framebuffer creation, lets store them in the framebuffer struct and then use them here. Then we'd only need to pass the buffer itself as an argument into this function